lints/lint_dnsname_wildcard_left_of_public_suffix.go

```diff
@@ -0,0 +1,54 @@
+package lints
+
+import (
+	"github.com/weppos/publicsuffix-go/publicsuffix"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type DNSNameWildcardLeftofPublicSuffix struct {
+	// Internal data here
+}
+
+func (l *DNSNameWildcardLeftofPublicSuffix) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameWildcardLeftofPublicSuffix) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.DNSNamesExist(c)
+}
+
+func wildcardLeftOfPublicSuffix(domain string) (bool, ResultStruct) {
```

Comment:
I don't think it makes sense to have ResultStruct at this point. Why don't you just return the err?

Reply:
I can do that and just check the value of err in the parent function, usually I don't propagate the error to higher level functions because it's harder to tell where the error is actually handled. What do you think? (This applies to the other lints as well)

Reply:
I'd prefer to have an error in a helper than a domain-specific struct.

Reply:
Got it.